mypy/test/testipc.py

```diff
@@ -44,12 +55,29 @@ def test_connect_twice(self) -> None:
         p.start()
         connection_name = queue.get()
         with IPCClient(connection_name, timeout=1) as client:
-            assert client.read() == msg.encode()
-            client.write(b"")  # don't let the server hang up yet, we want to connect again.
+            assert client.read() == msg
+            client.write("")  # don't let the server hang up yet, we want to connect again.
 
         with IPCClient(connection_name, timeout=1) as client:
-            assert client.read() == msg.encode()
-            client.write(b"test")
+            assert client.read() == msg
+            client.write("test")
+        queue.close()
+        queue.join_thread()
+        p.join()
+        assert p.exitcode == 0
+
+    def test_multiple_messages(self) -> None:
+        queue: Queue[str] = Queue()
+        p = Process(target=server_multi_message_echo, args=(queue,), daemon=True)
+        p.start()
+        connection_name = queue.get()
+        with IPCClient(connection_name, timeout=1) as client:
+            client.write("test1")
```

Comment:
Test writing whitespace and non-ascii characters? Maybe test all unicode code points within some range.

Reply:
Done! Also added a test for multiple consecutive messages to test when a buffer has multiple frames inside it.

Reply:
Though I didn't test all unicode code points. I tested some. I would trust codecs.encodeto base64 does the right thing and we don't have to worry about it.